### PR TITLE
Update project generation input validation

### PR DESCRIPTION
**Pull request by xorye**
This PR updates the input validation for when you go through the project generation wizard.

image

groupId: same as https://code.quarkus.io/
artifactId: same as https://code.quarkus.io/
project version: alphanumeric characters and these symbols: ._-
package name: same as https://code.quarkus.io/
resource name: must start with a letter (capital or lowercase), then, alphanumeric characters and these symbols: ._-

Signed-off-by: David Kwon <dakwon@redhat.com>

### CODE REVIEW

**Comment by xorye** [comment]
The regexes from the validation from https://code.quarkus.io/, are specified in this file: https://github.com/quarkusio/code.quarkus.io/blob/master/src/main/frontend/src/code-quarkus/pickers/__tests__/__snapshots__/info-picker.spec.tsx.snap

**Comment by fbricon** [comment]
That's good, but how do we handle server-side validation errors? We might lacking behind the server at some point, miss some validation. How are those server side errors managed?

**Comment by xorye** [comment]
If there is a server side error or a server side validation error, the request() fails and I am able to catch it. I'll implement a warning message

**Comment by xorye** [comment]
The following message appears when the server doesn't give us a project, which could happen due to many reasons (if we give it invalid parameters, if server is down, if we cannot connect to wifi etc.):

image

**Comment by xorye** [comment]
This PR is still not ready yet. I must continue testing. I'll post an update when things are good.

**Comment by xorye** [comment]
I think its good now. 

This PR now consists of two commits:
The first commit implements the error message that appears when a project cannot be downloaded from the endpoint for any reason.

The second commit implements client side validation for the inputs from the project generation wizard. It has been updated to match the new changes specified here: https://github.com/quarkusio/code.quarkus.io/pull/142#event-2632554602.

File: src/generateProject/validateInput.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  // https://github.com/quarkusio/code.quarkus.io/blob/master/src/main/frontend/src/code-quarkus/pickers/info-picker.tsx
+  const re = new RegExp('^([a-zA-Z_$][a-zA-Z\\d_$]*\\.)*[a-zA-Z_$][a-zA-Z\\d_$]*$');
+  if (!re.test(userInput)) {
+    return 'Invalid package name';
+  }
+  return undefined;
+}
+
+export async function validateResourceName(userInput: string): Promise<string|undefined> {
+  const re = new RegExp('^[A-Za-z]+[A-Za-z0-9_\\-.]*$');
```
Comment (fbricon) [inline src/generateProject/validateInput.ts]: `-` is not allowed in class names

**Review by fbricon — CHANGES_REQUESTED** [review]

**Pull request merged**